### ISSUE TITLE
The dimensionality seems weird 

### ISSUE THREAD

**Issue by mnzhao** [TencentAILab/tf-adaptive-softmax-lstm-lm#10]

**Comment by mnzhao** [comment]
When computing loss in adaptive softmax, we use head_logits ( with shape (sample_num, head_size)) and head_labels. However, the shape of head_labels seems to be of label dimension.

**Comment by mnzhao** [comment]
Also in line 68, it seems to me that with the tail_inputs and self.tail_w[i][0] cannot be multiplied since their dimension does not match?
